### PR TITLE
mds: check file layout in mknod

### PR DESCRIPTION
**Pull request by jhonxue**
Fixes: https://tracker.ceph.com/issues/64061
Signed-off-by: Xue Yantao <xueyantao2114@163.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by jhonxue** [comment]
@batrick  @vshankar PTAL

**Comment by jhonxue** [comment]
@tchaikov PTAL

**Comment by vshankar** [comment]
jenkins test api

File: src/mds/Server.cc
Comment (vshankar) [inline src/mds/Server.cc:7005]: Seems reasonable as the same check is performed in handle_client_openc(). Quick question - did you run into an issues due to the lack of this check? @jhonxue

Comment (jhonxue) [reply src/mds/Server.cc:7005]: Yes，encountered some situations, but not sure they are related. 

standby-reply mds:
           2024-01-11T07:56:13.731+0000 7f0077ca3700  0 mds.0.journal EMetaBlob.replay invalid layout on ino [inode 0x4003257a7c4 [2,head] #4003257a7c4 auth v1851322605 s=0 nl=0 n(v1079 rc2024-01-11T07:56:12.815778+0000 b67737072290 6452=6447+5) (iversion lock) 0x5610be080c00]: 000-1""
2024-01-11T07:56:13.732+0000 7f0077ca3700 -1 log_channel(cluster) log [ERR] : Invalid layout for inode 0x4003257a7c4 in journal
2024-01-11T07:56:14.640+0000 7f0077ca3700  1 mds.a.yj-bstoreceph01rack07meta-01.cmqeqz respawn!

Standby-reply mds report to  mon, mon blocklisted the active mds.

Active-mds :
             2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 mds.0.journaler.mdlog(rw) _finish_flush got (108) Cannot send after transport endpoint shutdown
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 mds.0.journaler.mdlog(rw) handle_write_error (108) Cannot send after transport endpoint shutdown
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 MDSIOContextBase: failed with -108, restarting...
2024-01-11T07:56:13.988+0000 7f1f6fc2d700  1 mds.a.yj-bstoreceph01rack08meta-01.jgtzrd respawn!

**Comment by dparmar18** [comment]
since layout is checked in many other functions; why not create a helper and call it wherever needed. I see the checks being the ditto same at every place.

File: src/mds/Server.cc
Comment (vshankar) [reply src/mds/Server.cc:7005]: If the standby-replay mds reports damaged to the mon, then the mon can blocklist the active mds. Did that happen?

Comment (jhonxue) [reply src/mds/Server.cc:7005]: Active-mds log:  active-mds can not flush journal to the rados

2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 mds.0.journaler.mdlog(rw) _finish_flush got (108) Cannot send after transport endpoint shutdown
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 mds.0.journaler.mdlog(rw) handle_write_error (108) Cannot send after transport endpoint shutdown
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 MDSIOContextBase: failed with -108, restarting...
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 1 mds.a.yj-bstoreceph01rack08meta-01.jgtzrd respawn!

**Comment by jhonxue** [comment]
> since layout is checked in many other functions; why not create a helper and call it wherever needed. I see the checks being the ditto same at every place.

good idea!!!

File: src/mds/Server.cc
Comment (jhonxue) [reply src/mds/Server.cc:7005]: mon log:  blocklisted mds

             2024-01-11T07:56:13.733+0000 7f89cb99e700  5 mon.yj-bstoreceph01rack05meta-01@0(leader).mds e286311 prepare_beacon mds.0 up:standby-replay -> down:damaged

2024-01-11T07:56:13.733+0000 7f89cb99e700  0 mon.yj-bstoreceph01rack05meta-01@0(leader).mds e286311 prepare_beacon: marking rank 0 damaged

2024-01-11T07:56:13.733+0000 7f89cb99e700  1 mon.yj-bstoreceph01rack05meta-01@0(leader).mds e286311 fail_mds_gid 1359886760 mds.a.yj-bstoreceph01rack07meta-01.cmqeqz role 0



bool MDSMonitor::fail_mds_gid(FSMap &fsmap, mds_gid_t gid)
{
  const auto& info = fsmap.get_info_gid(gid);
  dout(1) << "fail_mds_gid " << gid << " mds." << info.name << " role " << info.rank << dendl;

  ceph_assert(mon.osdmon()->is_writeable());

  epoch_t blocklist_epoch = 0;
  if (info.rank >= 0 && info.state != MDSMap::STATE_STANDBY_REPLAY) {
    utime_t until = ceph_clock_now();
    until += g_conf().get_val<double>("mon_mds_blocklist_interval");
    blocklist_epoch = mon.osdmon()->blocklist(info.addrs, until);
  }

  fsmap.erase(gid, blocklist_epoch);
  last_beacon.erase(gid);
  if (pending_daemon_health.count(gid)) {
    pending_daemon_health.erase(gid);
    pending_daemon_health_rm.insert(gid);
  }

  return blocklist_epoch != 0;
}

**Comment by dparmar18** [comment]
the order of commits should be inverse IMO and BTW if you're gonna use the helper, then there's no need of the first commit right

**Comment by jhonxue** [comment]
> the order of commits should be inverse IMO and BTW if you're gonna use the helper, then there's no need of the first commit right

yes

**Review by batrick — CHANGES_REQUESTED** [review]
Please change commit title to:

`mds: check file layout in mknod`

and please explain the reason: replay of the journal detects the invalid layout when replaying the journal.

@gregsfortytwo @vshankar file this one for justifying the s-r daemon causing the rank to go damaged. If there had been a failover, it would have done the same.

**Comment by gregsfortytwo** [comment]
> @gregsfortytwo @vshankar file this one for justifying the s-r daemon causing the rank to go damaged. If there had been a failover, it would have done the same.

Yes, but I'm more concerned that apparently an invalid layout got into the system at all. Did the directory layout get set wrong somehow (because it *is* gated!)? Did it copy improperly? Did a pool get set as the default data pool and then get deleted? (Oooh, that one is definitely possible.)

**Comment by jhonxue** [comment]
> > @gregsfortytwo @vshankar file this one for justifying the s-r daemon causing the rank to go damaged. If there had been a failover, it would have done the same.
> 
> Yes, but I'm more concerned that apparently an invalid layout got into the system at all. Did the directory layout get set wrong somehow (because it _is_ gated!)? Did it copy improperly? Did a pool get set as the default data pool and then get deleted? (Oooh, that one is definitely possible.)

There is only one data pool, and no delete. Affter we had done the fs disaster recovery  according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/,  this issue happened.

**Comment by jhonxue** [comment]
> > > > @gregsfortytwo @vshankar file this one for justifying the s-r daemon causing the rank to go damaged. If there had been a failover, it would have done the same.
> > > 
> > > 
> > > Yes, but I'm more concerned that apparently an invalid layout got into the system at all. Did the directory layout get set wrong somehow (because it _is_ gated!)? Did it copy improperly? Did a pool get set as the default data pool and then get deleted? (Oooh, that one is definitely possible.)
> > 
> > 
> > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> 
> You changed the PR title (which is good) but I'm asking you to fix the **commit** title.

yes, done.

**Comment by gregsfortytwo** [comment]
> There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.

Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?

**Comment by jhonxue** [comment]
> > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> 
> Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?

**Comment by jhonxue** [comment]
> > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > 
> > 
> > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?

Just replacing the journal.

> > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> 
> Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?

Just replacing the journal.

**Comment by dparmar18** [comment]
was this closed by mistake @jhonxue ?

**Comment by jhonxue** [comment]
> was this closed by mistake @jhonxue ?

Yes, thanks!

**Comment by vshankar** [comment]
> > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > 
> > > 
> > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> 
> Just replacing the journal.
> 
> > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > 
> > 
> > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> 
> Just replacing the journal.

We need a run down list of what happened. This is what I think happened:

- Invalid layout gets persisted in the journal due to missing check is mknod
- disaster recovery steps was performed
- mds rejoins the cluster - journal replay notices the invalid layout

Does it even matter if step(2) was executed? The recovery docs recommend to reset the journal. I guess that wasn't done? Hitting this likely requires multiple things to got wrong, but I'm not 100% clear about the chronology of events.

**Comment by jhonxue** [comment]
> > > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > > 
> > > > 
> > > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > 
> > 
> > Just replacing the journal.
> > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > 
> > > 
> > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > 
> > 
> > Just replacing the journal.
> 
> We need a run down list of what happened. This is what I think happened:
> 
> * Invalid layout gets persisted in the journal due to missing check is mknod
> * disaster recovery steps was performed
> * mds rejoins the cluster - journal replay notices the invalid layout
> 
> Does it even matter if step(2) was executed? The recovery docs recommend to reset the journal. I guess that wasn't done? Hitting this likely requires multiple things to got wrong, but I'm not 100% clear about the chronology of events.

we had done: 
1、stop all mds
2、cephfs-journal-tool --rank=a:all event recover_dentries summary
3、cephfs-journal-tool --rank=a:all journal reset

**Comment by vshankar** [comment]
> > > > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > > > 
> > > > > 
> > > > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > > 
> > > 
> > > Just replacing the journal.
> > > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > > 
> > > > 
> > > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > > 
> > > 
> > > Just replacing the journal.
> > 
> > We need a run down list of what happened. This is what I think happened:
> > 
> > * Invalid layout gets persisted in the journal due to missing check is mknod
> > * disaster recovery steps was performed
> > * mds rejoins the cluster - journal replay notices the invalid layout
> > 
> > Does it even matter if step(2) was executed? The recovery docs recommend to reset the journal. I guess that wasn't done? Hitting this likely requires multiple things to got wrong, but I'm not 100% clear about the chronology of events.
> 
> we had done: 
> 1、stop all mds
> 2、cephfs-journal-tool --rank=a:all event recover_dentries summary
> 3、cephfs-journal-tool --rank=a:all journal reset

Ok, so, recover_dentries wrote out the layout to the inode, which later resulted in the mds crashing. But, @gregsfortytwo question still remains - how did an invalid layout get set in the first place (without and pool deletes later).

**Comment by jhonxue** [comment]
> > > > > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > > > > 
> > > > > > 
> > > > > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > > > 
> > > > 
> > > > Just replacing the journal.
> > > > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > > > 
> > > > > 
> > > > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > > > 
> > > > 
> > > > Just replacing the journal.
> > > 
> > > 
> > > We need a run down list of what happened. This is what I think happened:
> > > 
> > > * Invalid layout gets persisted in the journal due to missing check is mknod
> > > * disaster recovery steps was performed
> > > * mds rejoins the cluster - journal replay notices the invalid layout
> > > 
> > > Does it even matter if step(2) was executed? The recovery docs recommend to reset the journal. I guess that wasn't done? Hitting this likely requires multiple things to got wrong, but I'm not 100% clear about the chronology of events.
> > 
> > 
> > we had done:
> > 1、stop all mds
> > 2、cephfs-journal-tool --rank=a:all event recover_dentries summary
> > 3、cephfs-journal-tool --rank=a:all journal reset
> 
> Ok, so, recover_dentries wrote out the layout to the inode, which later resulted in the mds crashing. But, @gregsfortytwo question still remains - how did an invalid layout get set in the first place (without and pool deletes later).

yes, this is also what bothers me. Currently, all I can do is prevent the invalida file layout event from entering the journal

**Comment by vshankar** [comment]
> > > > > > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > > > > > 
> > > > > > > 
> > > > > > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > > > > 
> > > > > 
> > > > > Just replacing the journal.
> > > > > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > > > > 
> > > > > > 
> > > > > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > > > > 
> > > > > 
> > > > > Just replacing the journal.
> > > > 
> > > > 
> > > > We need a run down list of what happened. This is what I think happened:
> > > > 
> > > > * Invalid layout gets persisted in the journal due to missing check is mknod
> > > > * disaster recovery steps was performed
> > > > * mds rejoins the cluster - journal replay notices the invalid layout
> > > > 
> > > > Does it even matter if step(2) was executed? The recovery docs recommend to reset the journal. I guess that wasn't done? Hitting this likely requires multiple things to got wrong, but I'm not 100% clear about the chronology of events.
> > > 
> > > 
> > > we had done:
> > > 1、stop all mds
> > > 2、cephfs-journal-tool --rank=a:all event recover_dentries summary
> > > 3、cephfs-journal-tool --rank=a:all journal reset
> > 
> > 
> > Ok, so, recover_dentries wrote out the layout to the inode, which later resulted in the mds crashing. But, @gregsfortytwo question still remains - how did an invalid layout get set in the first place (without and pool deletes later).
> 
> yes, this is also what bothers me. Currently, all I can do is prevent the invalida file layout event from entering the journal

OK. That's a separate effort and does not block this change. I'll run this through fs suite and report.

**Comment by jhonxue** [comment]
> > > > > > > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > > > > > 
> > > > > > 
> > > > > > Just replacing the journal.
> > > > > > > > There is only one data pool, and no delete. Affter we had done the fs disaster recovery according to the https://docs.ceph.com/en/latest/cephfs/disaster-recovery-experts/, this issue happened.
> > > > > > > 
> > > > > > > 
> > > > > > > Okay, I'm actually more worried about this. Did you end up replacing a pool? And not replacing the journal, I'm guessing?
> > > > > > 
> > > > > > 
> > > > > > Just replacing the journal.
> > > > > 
> > > > > 
> > > > > We need a run down list of what happened. This is what I think happened:
> > > > > 
> > > > > * Invalid layout gets persisted in the journal due to missing check is mknod
> > > > > * disaster recovery steps was performed
> > > > > * mds rejoins the cluster - journal replay notices the invalid layout
> > > > > 
> > > > > Does it even matter if step(2) was executed? The recovery docs recommend to reset the journal. I guess that wasn't done? Hitting this likely requires multiple things to got wrong, but I'm not 100% clear about the chronology of events.
> > > > 
> > > > 
> > > > we had done:
> > > > 1、stop all mds
> > > > 2、cephfs-journal-tool --rank=a:all event recover_dentries summary
> > > > 3、cephfs-journal-tool --rank=a:all journal reset
> > > 
> > > 
> > > Ok, so, recover_dentries wrote out the layout to the inode, which later resulted in the mds crashing. But, @gregsfortytwo question still remains - how did an invalid layout get set in the first place (without and pool deletes later).
> > 
> > 
> > yes, this is also what bothers me. Currently, all I can do is prevent the invalida file layout event from entering the journal
> 
> OK. That's a separate effort and does not block this change. I'll run this through fs suite and report.

OK.

**Comment by vshankar** [comment]
https://pulpito.ceph.com/vshankar-2024-02-17_05:57:56-fs-wip-vshankar-testing-20240217.015652-testing-default-smithi/

This run still has quay fetch failures

> Command failed on smithi155 with status 1: 'sudo /home/ubuntu/cephtest/cephadm --image quay-quay-quay.apps.os.sepia.ceph.com/ceph-ci/ceph:1c920afa8316ea21877f3186850254947531ea22 pull'

I'll file a tracker for that and merge this batch.

**Comment by vshankar** [comment]
> https://pulpito.ceph.com/vshankar-2024-02-17_05:57:56-fs-wip-vshankar-testing-20240217.015652-testing-default-smithi/
> 
> This run still has quay fetch failures
> 
> > Command failed on smithi155 with status 1: 'sudo /home/ubuntu/cephtest/cephadm --image quay-quay-quay.apps.os.sepia.ceph.com/ceph-ci/ceph:1c920afa8316ea21877f3186850254947531ea22 pull'
> 
> I'll file a tracker for that and merge this batch.

Unfortunately it was risky to merge the change without a proper run. The fix for this is under test.

**Comment by vshankar** [comment]
https://pulpito.ceph.com/?branch=wip-vshankar-testing-20240216.060239

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#26th-Feb-2024

**Comment by vshankar** [comment]
jenkins test make check

**Comment by batrick** [comment]
Maybe I'm lost in the conversation but if the question is **how** an invalid layout got in the journal... wouldn't it be via `mknod` (which this PR fixes)??

**Review by batrick — APPROVED** [review]

**Comment by gregsfortytwo** [comment]
> Maybe I'm lost in the conversation but if the question is **how** an invalid layout got in the journal... wouldn't it be via `mknod` (which this PR fixes)??

mknod simply inherits the layout from whichever directory ancestor has one set, and that layout was already checked prior to this change.

**Comment by jhonxue** [comment]
> > > Maybe I'm lost in the conversation but if the question is **how** an invalid layout got in the journal... wouldn't it be via `mknod` (which this PR fixes)??
> > 
> > 
> > mknod simply inherits the layout from whichever directory ancestor has one set, and that layout was already checked prior to this change.
> 
> I see.
> 
> @jhonxue something that's not clear to me: were the recovery steps performed before this invalid layout was found or only after? i.e. was any type of recovery ever done on the file system **before** the invalid layout crashed the MDS?
> 
> What version of Ceph are you running?

1. the recovery steps performed before this invalid layout was found 
2. 17.2.7

**Comment by batrick** [comment]
It seems to me the most likely explanation is that journal recovery introduced the invalid layout somehow. i.e. during `recover_dentries` the erroneous layout was written to whatever directory inode which was inherited by a future `mknod`.

**Comment by jhonxue** [comment]
> It seems to me the most likely explanation is that journal recovery introduced the invalid layout somehow. i.e. during `recover_dentries` the erroneous layout was written to whatever directory inode which was inherited by a future `mknod`.

I agree with your explanation

**Comment by vshankar** [comment]
> It seems to me the most likely explanation is that journal recovery introduced the invalid layout somehow. i.e. during `recover_dentries` the erroneous layout was written to whatever directory inode which was inherited by a future `mknod`.

Given that this was hit with 17.2.7 bothers me (the bug is still lingering around in supported versions o_O).

**Comment by vshankar** [comment]
jenkins test make check

**Comment by vshankar** [comment]
jenkins test make check

**Comment by vshankar** [comment]
For some reason I cannot trigger make check which is a required check to merge this change.

@jhonxue ping rebasing locally and pushing please?

**Comment by jhonxue** [comment]
> For some reason I cannot trigger make check which is a required check to merge this change.
> 
> @jhonxue ping rebasing locally and pushing please?

ok，thanks

**Comment by jhonxue** [comment]
@vshankar  ceph API tests failed.  already reported issues

Python Unit-test Error

ERROR: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)

⇒ Check for [already reported issues](https://tracker.ceph.com/search?q=ERROR:%20test_list_enabled_module%20%28tasks.mgr.dashboard.test_mgr_module.MgrModuleTest%29&open_issues=1) on "ERROR: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)" in Ceph Tracker...

**Comment by vshankar** [comment]
> @vshankar ceph API tests failed. already reported issues
> 
> Python Unit-test Error
> 
> ERROR: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
> 
> ⇒ Check for [already reported issues](https://tracker.ceph.com/search?q=ERROR:%20test_list_enabled_module%20%28tasks.mgr.dashboard.test_mgr_module.MgrModuleTest%29&open_issues=1) on "ERROR: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)" in Ceph Tracker...

Thx. I've reported this in the tracker. BTW, the API tests are mandatory for merging changes.

**Comment by vshankar** [comment]
jenkins test api

**Comment by vshankar** [comment]
jenkins test make check

**Comment by vshankar** [comment]
jenkins test api

**Comment by batrick** [comment]
For what it's worth, I spent some time auditing the journal tool but didn't find the cause. I did discover this however: https://tracker.ceph.com/issues/64602

**Comment by vshankar** [comment]
> For what it's worth, I spent some time auditing the journal tool but didn't find the cause. I did discover this however: https://tracker.ceph.com/issues/64602

Fair point - I'll have someone dive into the details in a separate effort.

**Pull request merged**
